src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java

```diff
@@ -153,9 +167,31 @@ private CredentialsProvider getApiCredentialsProvider(
 
         provider.setCredentials(AuthScope.ANY, creds);
 
+        // Set proxy credentials if provided
+        if (proxy != null && !StringUtils.isBlank(proxy.getUserName())) {
```

Comment:
Cloud blank usernames be considered value for someone's proxy configuration? (e.g. password only)?

Reply:
I have not seen any case like this, but I will update the code. The UsernamePasswordCredentials class does not allow null values for the username, but I can pass in an empty string.